torch/_inductor/constant_folding.py

```diff
@@ -118,7 +118,13 @@ def set_env(arg):
         args, kwargs = self.fetch_args_kwargs_from_env(node)
         flattened_inputs = pytree.arg_tree_leaves(*args, **kwargs)
 
-        if self.unknown_value in flattened_inputs:
+        # We need to do this weird thing because in cases where flattened_inputs
+        # contains a ScriptObject, equality checking results in a type error if
+        # the types are different.
```

Comment:
Is this something that can be fixed on ScriptObject ?

Reply:
I'm not sure how? Because if object the equality check was type-erroring on had type "object". I'm not even sure what kind of object that was...

Reply:
Could we return false in the case that we would throw ? maybe file an issue for this ?